### PR TITLE
Add details about k8s annotations

### PR DESCRIPTION
**Pull request by sblausten**
https://app.shortcut.com/larder/story/15368/bailliegifford-issues-with-k8s-frontend-displaying-pods

### CODE REVIEW

File: content/docs/integrations/kubernetes/index.md [hunk 1]
```diff
@@ -22,7 +22,58 @@ Roadie supports the following connection setups:
...
+
+```yaml
+---
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  ...
+  annotations:
+    backstage.io/kubernetes-id: something
+    backstage.io/kubernetes-namespace: roadie
```
Comment (dtuite) [inline content/docs/integrations/kubernetes/index.md:44]: Reading to this point, I still don't understand the value of the `backstage.io/kubernetes-namespace` label. Is there something else we can say to explain it? If I remove this, what part of the plugin stops working?

Comment (sblausten) [reply content/docs/integrations/kubernetes/index.md:44]: I don't either tbh. Its just an additional layer I guess, maybe if you want to keep the id or labels generic then you may need a namespace label to specify exactly what you want to show up and avoid id clashes.

Comment (sblausten) [reply content/docs/integrations/kubernetes/index.md:44]: I clarified this - I was actually wrong initially on the usage

File: content/docs/integrations/kubernetes/index.md [hunk 2]
```diff
@@ -22,7 +22,60 @@ Roadie supports the following connection setups:
 
 All three of these can use a Broker service that runs in your infrastructure and means that you do not have to expose your cluster via an AWS cross account Role, Google OAUTH App token or Service Account Token.
 
-# Via Broker
+# Kubernetes Annotations
+
+The k8s plugin uses two alternate ways of discovering resources in addition to a namespace annotation. You can find the Backstage docs on this [here](https://backstage.io/docs/features/kubernetes/configuration/#surfacing-your-kubernetes-components-as-part-of-an-entity).
+
+## 1. backstage.io/kubernetes-id
+You can add a label with the key `backstage.io/kubernetes-id` to all of your k8s resources relating to an entity and then using the same key as an annotation on your entity.
```
Comment (iain-b) [inline content/docs/integrations/kubernetes/index.md:30]: I'm assuming the value is arbitrary?

File: content/docs/integrations/kubernetes/index.md [hunk 3]
```diff
@@ -22,7 +22,60 @@ Roadie supports the following connection setups:
...
+
+Additionally, you can lookup resources by namespace using the `backstage.io/kubernetes-namespace` annotation.
```
Comment (iain-b) [inline content/docs/integrations/kubernetes/index.md:32]: I'm assuming the default is "default"

File: content/docs/integrations/kubernetes/index.md [hunk 4]
```diff
@@ -22,7 +22,60 @@ Roadie supports the following connection setups:
...
+kind: Component
+metadata:
+  ...
+  annotations:
+    backstage.io/kubernetes-id: something
+    backstage.io/kubernetes-namespace: roadie
+...
+```
+
+```yaml
```
Comment (iain-b) [inline content/docs/integrations/kubernetes/index.md:48]: perhaps a comment just to highlight that this is a k8s resource and the above is a Backstage entity ? They have the same structure so I can see how users might get confused.

**Review by iain-b — APPROVED** [review]
Lgtm with some nit picks

File: content/docs/integrations/kubernetes/index.md (hunk 2 above)
Comment (sblausten) [reply content/docs/integrations/kubernetes/index.md:30]: Yes in their example they use the entity name but it just searches as if it were any other label.

File: content/docs/integrations/kubernetes/index.md (hunk 3 above)
Comment (sblausten) [reply content/docs/integrations/kubernetes/index.md:32]: Oh well I just tested that and I think I'm wrong there.

Comment (sblausten) [reply content/docs/integrations/kubernetes/index.md:32]: No that was just a misconfig

**Pull request merged**